draft-wang-ppm-dap-taskprov.md

```diff
@@ -570,13 +570,19 @@ Author to be under control of the adversary. It is therefore incumbent on
 protocol participants to verify the privacy parameters of a task before opting
 in.
 
```

Comment:
It's not clear to me why tracking a client across uploads leaks the client's location. Do you have an idea in mind? 

I realize this is mentioned in #48, but now I'm not sure it's the best example. Stepping back, what does an attacker that controls the Author (and possible the Collector and an Aggregator) by learn by fingerprinting a Client's upload task? Perhaps this would violate other properties adjcent to DAP, like anonymity or differential privacy?

Reply:
I think if a client sends report to leader directly, then the upload will contain things like IP, or user agent, etc. That can leak the client location, device. And by not checking task parameters carefully, as a task author, I can set min_batch_size to be 1, and see the client measurement directly, along with user agent and IP.

Reply:
> I think if a client sends report to leader directly, then the upload will contain things like IP, or user agent, etc.

Right: DAP on its own doesn't provide for anonymity. Suppose though you rand DAP on top of OHTTP.

> That can leak the client location, device. And by not checking task parameters carefully, as a task author, I can set min_batch_size to be 1, and see the client measurement directly, along with user agent and IP.

This is supposed to be prevented by the "opt-in" procedure.

Reply:
I guess maybe looking at this issue again, do we just care about the capability of author colluding with leader, where author is not a collector? So it cannot see the measurement in the clear. So you can only get the metadata information of a particular client, e.g. IP and user agent. Is that the only security risk we care about then?

Reply:
Yeah I think that's a good example of the kind of risk we're thinking of. More to the point: The metadata may convey to the Leader who the client is; and if you're expecting that to be impossible, say because you use OHTTP, then you have to be very careful about how taskprov is deployed.

Reply:
But I think we could also make the point that, even if the attacker can deanonymize a client, it's hard for it do learn anything more than the client uploaded a report. If the task is unique to that client, and the batch size is larger than 1, then it's really not going to get any more information.